chrome/content/zotero/preferences/storageBreakdown.jsx

```diff
@@ -0,0 +1,182 @@
+/*
+    ***** BEGIN LICENSE BLOCK *****
+
+    Copyright © 2021 Center for History and New Media
+                     George Mason University, Fairfax, Virginia, USA
+                     http://zotero.org
+
+    This file is part of Zotero.
+
+    Zotero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+
+    ***** END LICENSE BLOCK *****
+*/
+'use strict';
+
+import React, { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { IntlProvider, useIntl } from "react-intl";
+import { humanReadableSize } from 'components/utils';
+
+Components.utils.import("resource://gre/modules/osfile.jsm");
+
+const StorageBreakdown = () => {
+	const intl = useIntl();
+
+	const [loading, setLoading] = useState(true);
+	const [partitions, setPartitions] = useState([]);
+
+	useEffect(() => {
+		const calculateStorage = async () => {
+			let libraries = Zotero.Libraries.getAll().map((library) => {
+				return {
+					libraryID: library.libraryID,
+					name: library.name,
+					size: 0,
+					count: 0
+				};
+			});
+
+			setPartitions(libraries.map(library => library));
+			let update = 0;
+
+			await Zotero.Promise.all(libraries.map(async (partition, index) => {
+				let items = await Zotero.Items.getAll(partition.libraryID);
+
+				const getFileSize = async (attachmentFile) => {
+					if (!attachmentFile.name.startsWith('.')) {
+						try {
+							let size = (await OS.File.stat(attachmentFile.path)).size;
+							libraries[index].size += size;
+						}
+						catch (e) {
+							if (e instanceof OS.File.Error && e.becauseNoSuchFile) {
+								// File may or may not exist on disk, but we
+								// don't care so swallow this error
+							}
+							else {
+								throw e;
```

Comment:
Instead of throwing, maybe just `Zotero.logError(e)` and ignore? One way or another, we should account for the possibility of permissions or I/O errors and not break the dialog if those occur.